java/maven/src/org/netbeans/modules/maven/execute/MavenCommandLineExecutor.java

```diff
@@ -100,7 +103,35 @@
 
 /**
  * support for executing maven, externally on the command line.
+ * <b>Since 2/1.144</b>, the {@link LateBoundPrerequisitesChecker} registered in Maven projects by default supports 
+ * {@link ExplicitProcessParameters} API. The caller of the execute-type action can request to append or replace VM or user
+ * application parameters. The parameters recorded in the POM.xml or NetBeans action mappings are augmented according to that
+ * instructions:
+ * <ul>
+ * <li><b>priorityArgs</b> are mapped to VM arguments (precede main class name)
+ * <li><b>args</b> are mapped to user application arguments (after main class name)
+ * </ul>
+ * VM parameters injected by {@link StartupExtender} API are not affected by this feature. 
+ * <p>
+ * Example use:
+ * <code>
+ *   ActionProvider ap = ... ; // obtain ActionProvider from the project.
```

Comment:
If this could go into the documentation of `ExplicitProcessParameters` (even as a `non-normative` section), it would be far more visible. I am afraid Javadoc of this class isn't published in the official release documentation.

Reply:
Documented in 39417f0d8b